### ISSUE TITLE
Clarify IE11 support state for `zone.js` / and clean-up zone.js code/tests for IE11-specific logic

### ISSUE THREAD

**Issue by devversion** [angular/angular#46773]
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

No

### Description

Angular no longer supports IE11, but it's not quite clear from the documentation whether `zone.js` is included in the angular.io browser guarantees, or if Zone has its own browser guarantees (given it also being versioned differently etc.)

If IE11 is no longer supported in ZoneJS either, ZoneJS tests and logic should be cleaned up to not contain any IE11 specific code. There are some IE11-specific tests for example though (these don't even run at this point anymore -- worth mentioning)

**Comment by JiaLiPassion** [comment]
@devversion , yes, there are still some test cases and legacy support logic to support `IE11` and old versionf of Safari, Edge and some Android Browsers. If the policy is decided, I will clean up these logic.
